### PR TITLE
oci-register-machine will not terminate containers with ID > 32 chars

### PR DESCRIPTION
**Pull request by rhatdan**
This patch truncates the ID the same way in Terminate as it does in
register.

### CODE REVIEW

**Comment by rhatdan** [comment]
@mrunalp @sallyom PTAL

File: oci-register-machine.go [hunk 1]
```diff
@@ -79,6 +79,10 @@ func TerminateMachine(name string) error {
 			return err
 		}
 	}
+	if len(name) > 32 {
```
Comment (mrunalp) [inline oci-register-machine.go]: This is problematic as it could lead to clashes. I think we might have to use the ID here.

Comment (rhatdan) [inline oci-register-machine.go]: Ok because your names are not unique until the end.

**Comment by rhatdan** [comment]
@giuseppe PTAL

Why did you change to only do the Validate if the string is less then 32

File: oci-register-machine.go (hunk 1 above)
Comment (mrunalp) [inline oci-register-machine.go]: Yes.

**Comment by giuseppe** [comment]
I was assuming that if the string is at least 32 chars (as per the previous check where I read the uuid from the container) it is already a valid hex.  We can probably drop that check safely.  RegisterMachine is already assuming the maximum length to be 32 though that is what RegisterMachine expects:

https://www.freedesktop.org/wiki/Software/systemd/machined/

**Comment by rhatdan** [comment]
Anyways @mrunalp and @giuseppe does this patch LGTY?

**Comment by giuseppe** [comment]
@rhatdan LGTM.  Should the issue reported by @mrunalp be addressed somehow?

**Comment by mrunalp** [comment]
@giuseppe We addressed that cri-o with https://github.com/kubernetes-incubator/cri-o/pull/629

File: oci-register-machine.go [hunk 2]
```diff
@@ -143,11 +147,9 @@ func main() {
...
 	}
 	// ensure id is a hex string at least 32 chars
-	if len(passId) < 32 {
-		passId, err = Validate(passId)
-		if err != nil {
-			log.Fatalf("RegisterMachine Failed %v", err.Error())
-		}
+	passId, err = Validate(passId)
+	if err != nil {
+		log.Fatalf("RegisterMachine Failed %v", err.Error())
```
Comment (mrunalp) [inline oci-register-machine.go]: nit: This should just be `err` not `err.Error()`

**Comment by rhatdan** [comment]
@giuseppe they have been addressed in CRI-O, which has been merged.

**Comment by rhatdan** [comment]
Added another patch for @mrunalp Nit.  Everything else is ready to go, I believe.

**Comment by giuseppe** [comment]
LGTM.  Unfortunately I can't merge it as I have no commit access and it seems that Homu is not enabled here

**Comment by filbranden** [comment]
@rhatdan Regarding your commit "Use %m for handling output of err.Error()", I don't think that works as intended...

I've been testing some changes on a machine that make oci-register-machine fail (if you're curious, I've been running dockerd with cgroupfs driver instead of systemd driver) and, when it fails, what it logs is:

```
Jan 10 01:15:13 testhost oci-register-machine[17034]: 2018/01/10 01:15:13 Register machine failed: %!m(MISSING)
```

I imagine what you intended here is what %m does in printf in C which is print the message from errno, but I don't think Go supports that...

May I suggest reverting that commit?

Thanks!
Filipe

**Pull request merged**